Modules/Microsoft365DSC/DSCResources/MSFT_EXOPhishSimOverrideRule/MSFT_EXOPhishSimOverrideRule.psm1

```diff
@@ -0,0 +1,435 @@
+        [string]$propertyName
+    )
+
+    # Get the arrays
+    $setArray = $setParameters[$propertyName]
+    $currentArray = $currentInstance[$propertyName]
+
+    # Compare arrays
+    $addArray = $setArray | Where-Object { $_ -notin $currentArray }
+    $removeArray = $currentArray | Where-Object { $_ -notin $setArray }
+
+    # Modify $setParameters
+    if ($addArray.Count -gt 0) {
+        $setParameters.Add("Add$propertyName", $addArray)
+    }
+    if ($removeArray.Count -gt 0) {
+        $setParameters.Add("Remove$propertyName", $removeArray)
+    }
+
+    # Remove the original property
+    $setParameters.Remove($propertyName)
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Identity,
+
+        [Parameter()]
+        [System.String[]]
+        $Domains,
+
+        [Parameter()]
+        [System.String[]]
+        $SenderIpRanges,
+
+        [Parameter()]
+        [System.String]
+        $Comment,
+
+        [Parameter()]
+        [System.String]
+        $Policy,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint,
+
+        [Parameter()]
+        [Switch]
+        $ManagedIdentity,
+
+        [Parameter()]
+        [System.String[]]
+        $AccessTokens
+    )
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName.Replace('MSFT_', '')
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $currentInstance = Get-TargetResource @PSBoundParameters
+
+    $setParameters = Remove-M365DSCAuthenticationParameter -BoundParameters $PSBoundParameters
+
+    # CREATE
+    if ($Ensure -eq 'Present' -and $currentInstance.Ensure -eq 'Absent')
+    {
+        $ruleIdentity = $setParameters['Identity']
+        $setParameters.Add("Name", $ruleIdentity)
+        $setParameters.Remove("Identity")
+
+        New-EXOPhishSimOverrideRule @SetParameters
+    }
+    # UPDATE
+    elseif ($Ensure -eq 'Present' -and $currentInstance.Ensure -eq 'Present')
+    {
+        # Modify Domains and SenderIpRanges parameters as Set cmdlet for this resource has different parameter names
+        ModifyPropertiesForSetCmdlet -setParameters $setParameters -currentInstance $currentInstance -propertyName "Domains"
```

Comment:
I believe these would need to be assigned as variable, or passed as reference. Calling ModifyPropertiesForSetCmdlet will only change the value in the methods scope but once it returns to the Set-TargetResource context, those changes are lost.

Reply:
By default, parameters are passed as reference in PowerShell. I have debugged and verified